src/OnlineSales/Data/ApiDbContext.cs

```diff
@@ -94,16 +94,18 @@ public override async Task<int> SaveChangesAsync(bool acceptAllChangesOnSuccess,
         {
             if (entityEntry.State == EntityState.Modified)
             {
-                ((BaseEntity)entityEntry.Entity).UpdatedAt = DateTime.UtcNow;
-                ((BaseEntity)entityEntry.Entity).UpdatedByIp = httpContextHelper!.IpAddress;
-                ((BaseEntity)entityEntry.Entity).UpdatedByUserAgent = httpContextHelper!.UserAgent;
+                var entity = (BaseEntity)entityEntry.Entity;
+                entity.UpdatedAt = entity.UpdatedAt is null ? DateTime.UtcNow : entity.UpdatedAt;
```

Comment:
@gsplakmal will not it be only updated during the first update and then always stay the same?

To ensure the logic stays consistent after this modification please create a test to verify that UpdateAt date is always update after each entity instance update (PATCH).